### PR TITLE
feat: using signed URL to upload file

### PR DESCRIPTION
**Pull request by VB-at-Bis**
Implements #982

### CODE REVIEW

File: app/backend/lib/s3client.ts [hunk 1]
```diff
@@ -32,9 +34,22 @@ aws.config.update({
 const s3Client = new aws.S3();
 
 export default s3Client;
-
+const agent = new https.Agent({
```
Comment (VB-at-Bis) [inline app/backend/lib/s3client.ts:37]: additional config settings to prevent 'Socket hung up' errors.

File: app/backend/lib/s3client.ts [hunk 2]
```diff
@@ -32,9 +34,22 @@ aws.config.update({
...
+  maxSockets: 300,
+  keepAlive: true
+ });
 const awsConfig = {
   region: AWS_S3_REGION,
+  logger: console,
+  httpOptions: {
+    timeout: 45000,
```
Comment (VB-at-Bis) [inline app/backend/lib/s3client.ts:45]: additional config settings to prevent 'Socket hung up' errors.

**Review by marcellmueller — APPROVED** [review]
This is working really well, nice job. 

I think we should do some load testing today just to be safe. Up to you if you want to merge this first then test or deploy this in load testing mode first.

**Comment by VB-at-Bis** [comment]
Load tests executed with no issues:
https://github.com/bcgov/CONN-CCBC-portal/actions/runs/3857769425

**Pull request merged**